src/ACEtk/interpretation/ContinuousEnergyNeutron/Builder/EnergyDistribution/KalbachMann/Data/Builder/test/Builder.test.cpp

```diff
@@ -0,0 +1,124 @@
+         .angularDistributionSlope( njoy::utility::copy( A ) );
+
+    auto distribution = dataB.construct();
+    THEN( "the values can be verified" ){
+      CHECK( INTT == distribution.interpolationParameter );
+      CHECK( ranges::equal( ene, distribution.energies ) );
+      CHECK( ranges::equal( pdf, distribution.pdf ) );
+      CHECK( ranges::equal( cdf, distribution.cdf ) );
+      CHECK( ranges::equal( R, distribution.precompoundFraction ) );
+      CHECK( ranges::equal( A, distribution.angularDistributionSlope ) );
+
+      AND_THEN( "the contents can be ACE-ified" ){
+        std::vector< double > aceified{};
+        aceified.push_back( INTT );
+        aceified.push_back( ene.size() );
+        aceified |= ranges::action::push_back( ene );
+        aceified |= ranges::action::push_back( pdf );
+        aceified |= ranges::action::push_back( cdf );
+        aceified |= ranges::action::push_back( R );
+        aceified |= ranges::action::push_back( A );
+
+        Table::Data data{};
+        distribution.ACEify( data, 0 );
+
+        CHECK( ranges::equal( aceified, data.XSS() ) );
+      }
+    }
+  } // GIVEN valid
+  GIVEN( "invalid inputs" ){
+    WHEN( "energies are negative" ){
+      std::vector< double > energies{ -1.0, 2.0, 5.0, 6.0 };
+
+      THEN( "an exception is thrown" ){
+        CHECK_THROWS_AS( 
+          dataB.energies( njoy::utility::copy( energies ) ),
+          details::verify::exceptions::NotPositive&
+        );
+      }
+    }
+    WHEN( "energies are not sorted" ){
+      std::vector< double > energies{ 1.0, 2.0, 5.0, 4.0 };
+
+      THEN( "an exception is thrown" ){
+        CHECK_THROWS_AS( 
+          dataB.energies( njoy::utility::copy( energies ) ),
+          details::verify::exceptions::Unsorted&
+        );
+      }
+    }
+    WHEN( "pdf values are negative" ){
+      std::vector< double > pdf{ -1.0, 2.0, 5.0, 6.0 };
+
+      THEN( "an exception is thrown" ){
+        CHECK_THROWS_AS( 
+          dataB.pdf( njoy::utility::copy( pdf ) ),
+          details::verify::exceptions::NotPositive&
+        );
+      }
+    }
+    WHEN( "cdf values are negative" ){
+      std::vector< double > cdf{ -1.0, 2.0, 5.0, 6.0 };
+
+      THEN( "an exception is thrown" ){
+        CHECK_THROWS_AS( 
+          dataB.cdf( njoy::utility::copy( cdf ) ),
+          details::verify::exceptions::InvalidCDF&
+        );
+      }
+    }
+    WHEN( "cdf values are not sorted" ){
+      std::vector< double > cdf{ 1.0, 2.0, 5.0, 4.0 };
+
+      THEN( "an exception is thrown" ){
+        CHECK_THROWS_AS( 
+          dataB.cdf( njoy::utility::copy( cdf ) ),
+          details::verify::exceptions::InvalidCDF&
+        );
+      }
+    }
+    WHEN( "interpolation parameter is <= 0 " ){
+      THEN( "an exception is thrown" ){
+        CHECK_THROWS( dataB.interpolationParameter(  0 ) );
+        CHECK_THROWS( dataB.interpolationParameter( -1 ) );
+      }
+    }
+  } // GIVEN invalid
```

Comment:
Adapt test as required following introduction of numberDiscretePhotons

Reply:
Done